src/lib.rs

```diff
@@ -433,178 +431,224 @@ mod tests {
     ];
 
     #[derive(Default)]
-    struct OscDispatcher {
-        dispatched_osc: bool,
-        bell_terminated: bool,
-        params: Vec<Vec<u8>>,
+    struct Dispatcher {
+        dispatched: Vec<Sequence>,
     }
 
-    // All empty bodies except osc_dispatch
-    impl Perform for OscDispatcher {
-        fn osc_dispatch(&mut self, params: &[&[u8]], bell_terminated: bool) {
-            // Set a flag so we know these assertions all run
-            self.dispatched_osc = true;
-            self.bell_terminated = bell_terminated;
-            self.params = params.iter().map(|p| p.to_vec()).collect();
-        }
+    #[derive(Debug, PartialEq, Eq)]
+    enum Sequence {
+        Osc(Vec<Vec<u8>>, bool),
+        Csi(Vec<Vec<u16>>, Vec<u8>, bool, char),
+        Esc(Vec<u8>, bool, u8),
+        DcsHook(Vec<Vec<u16>>, Vec<u8>, bool, char),
+        DcsPut(u8),
```

Comment:
Would these be better served as `{}` struct variants of the enum to help make it clear what the fields mean?

Reply:
I've considered that, but it's test code so I didn't really care that much to be honest.

Depends on how you work with your code though I suppose. For me that makes no difference, but for some people it would probably be easier to understand.

Reply:
Yea, I mean it's documented in the deconstructing I suppose. I just often like to read definitions to get a sense of what's going on.